scripts/fast-reboot

```diff
@@ -271,14 +271,23 @@ function teardown_control_plane_assistant()
     fi
 }
 
+function is_secureboot() {
+    grep -Eq 'secure_boot_enable=[1y]' /proc/cmdline
+}
+
 function setup_reboot_variables()
 {
     # Kernel and initrd image
     NEXT_SONIC_IMAGE=$(sonic-installer list | grep "Next: " | cut -d ' ' -f 2)
     IMAGE_PATH="/host/image-${NEXT_SONIC_IMAGE#SONiC-OS-}"
     if grep -q aboot_platform= /host/machine.conf; then
-        KERNEL_IMAGE="$(ls $IMAGE_PATH/boot/vmlinuz-*)"
-        BOOT_OPTIONS="$(cat "$IMAGE_PATH/kernel-cmdline" | tr '\n' ' ') SONIC_BOOT_TYPE=${BOOT_TYPE_ARG}"
+        if is_secureboot; then
+            KERNEL_IMAGE=""
+            BOOT_OPTIONS="SONIC_BOOT_TYPE=${BOOT_TYPE_ARG}"
```

Comment:
>" [](start = 58, length = 1)

Will you add `secure_boot_enable=1` into `BOOT_OPTIONS`?

This is also applied to non-aboot secure boot flow.

Reply:
Because the loading of the kernel and initramfs is done through boot0 the `secure_boot_enable=1` option is added automatically. Adding it here would be redundant but harmless.
Let me know if you still want me to add it here.

Reply:
Please help added it. This code will be shared to non-aboot secure boot flow which using `kexec`.

I guess the aboot has the intelligence to keep single `secure_boot_enable=1` if there is already one. There is no need to duplicate it on cmdline.

---
In reply to: [456649028](https://github.com/Azure/sonic-utilities/pull/994#discussion_r456649028) [](ancestors = 456649028)